mdoc/mdoc.Test/CppFullFormatterTests.cs

```diff
@@ -0,0 +1,328 @@
+﻿using System;
+using Mono.Cecil;
+using NUnit.Framework;
+using Mono.Documentation.Updater.Formatters.CppFormatters;
+using Mono_DocTest_Generic;
+using Cpp=Mono_DocTest_Generic;
+using Mono_DocTest;
+
+namespace mdoc.Test
+{
+    public class CppFullFormatterTests: BasicFormatterTests<CppFullMemberFormatter>
+    {
+        private static readonly CppFullMemberFormatter cppFullMemberFormatter = new CppFullMemberFormatter();
+        protected override CppFullMemberFormatter formatter => cppFullMemberFormatter;
+        private const string CSharpTestLib = "CSharpExample.dll";
+        [Test]
+        [Category("Methods")]
+        public void MethodSignature_Bar() =>
+            TestMethodSignature(typeof(Cpp.Extensions), @"public:
```

Comment:
I'm seeing an inconsistency between, for example, the visibility modifier here (there's a newline), and the ones in `CppFormatterTests.cs`, which do not feature a newline. Can you talk about what's going on here? 

Additionally, I'm a tiny bit concerned with the use of the `@` string literal to insert the newline here ... while it's working right now, a future formatting by a maintainer could introduce platform-dependent newline characters, rather than the standardized `\n`. I'd rather see the explicit use of of `\n` in a string declared without `@`. This makes the intent clearer, and less prone to confusion in the future.

Reply:
Talking about the modifiers, when you do a type declaration then visibility modifier is added without any punctuation signs and is placed in one line. When you declare type members (including nested types), then  visibility modifier is added with ":". For methods/fields/properties visibility modifier can be put on different line. It means that modifier will be applied for the members defined on next lines.
Documentation uses the same approach (see code examples via the [link] (https://msdn.microsoft.com/en-us/library/2f1ec0b1.aspx))

Reply:
ahh I see ... so then the discrepancy I noticed was just types vs members. ok, that's reasonable.